File: q09_arraymult.go
package main

import "fmt"

func simple(arr []int) []int {
	ret := []int{}
	num := 1
	for _, n := range(arr) {
		num *= n
	}
	for _, n := range(arr) {
		ret = append(ret, num/n)
	}
	return ret
}

func main() {
	arr := []int{1, 2, 3, 4, 5}
	ret := simple(arr)
	for _, n := range(ret) {
		fmt.Print(n, " ");
	}
}

File: q09_arraymult.cpp
#include <iostream>

void simple(int* arr, int len, int* ret) {
    int num = 1;
    for (auto i = 0; i < len; i++)
        num *= arr[i];
    for (auto i = 0; i < len; i++)
        ret[i] = num/arr[i];
}

int main() {
    int arr[5] {1,2,3,4,5};
    int ret[5] {0,0,0,0,0};
    simple(arr, 5, ret);

    for (auto i : ret)
        std::cout << i << " ";
    std::cout << "\n";
    return 0;
}
